src/internal/document_api/mod.rs

```diff
@@ -250,10 +254,80 @@ impl DocumentMetadataResult {
     }
 }
 
+/// Result for encrypt operations that do not store document access information with the webservice,
+/// but rather return the access information as `encrypted_deks`. Both the `encrypted_data` and
+/// `encrypted_deks` must be used to decrypt. See `document_edek_decrypt`
+///
+/// - `id` - Unique (within the segment) id of the document
+/// - `encrypted_data` - Bytes of encrypted document content
+/// - `encrypted_deks` - List of encrypted document encryption keys (EDEK) of users/groups that have been granted access to `encrypted_data`
+/// - `access_errs` - Users and groups that could not be granted access
+#[derive(Debug)]
+pub struct DocumentEncryptUnmanagedResult {
+    id: DocumentId,
+    encrypted_data: Vec<u8>,
+    encrypted_deks: Vec<u8>,
+    access_errs: Vec<DocAccessEditErr>,
+}
+
+impl DocumentEncryptUnmanagedResult {
+    pub fn id(&self) -> &DocumentId {
+        &self.id
+    }
+    pub fn encrypted_data(&self) -> &[u8] {
+        &self.encrypted_data
+    }
+    pub fn encrypted_deks(&self) -> &[u8] {
+        &self.encrypted_deks
+    }
+    pub fn access_errs(&self) -> &[DocAccessEditErr] {
+        &self.access_errs
+    }
+
+    /// Users and Groups that have access to the encrypted document.
+    ///
+    /// Implementation is somewhat expensive as it is decoding the proto encoded `encrypted_deks`.
+    pub fn grants(&self) -> Result<Vec<UserOrGroup>, IronOxideErr> {
```

Comment:
I ended up just caching the grants on the result creation -- which was what I was originally trying to avoid. Deserializing proto just makes the interface too ugly...